### PR TITLE
More doxygen commenting of stat and skill effects [CR]

### PR DESCRIPTION
**Pull request by sparr**
I'm about 25% of the way through all the stat effects. I'll probably get the rest in another couple of commits. I'll also be doing skill effects in this patch.

Comments on formatting and verbiage are welcome. Additional doxygen scripts/configuration would be helpful as well.

To produce the documentation, run `doxygen doxygen_doc/doxygen_conf.txt` from your Cataclysm-DDA repo folder and then browse to doxygen_doc/html/pages.html

### CODE REVIEW

**Comment by sparr** [comment]
I think I've got everywhere that stats are used. I'd appreciate if someone could check all the STAT_EFFECT_ comments and make sure they at least seem to describe what the code is doing.

**Comment by narc0tiq** [comment]
> To produce the documentation, [...]

Also, once this is merged, you will see the updates at <http://dev.narc.ro/cataclysm/doxygen/>. I'm honestly tempted to manually update it with the stuff in this branch, but it would just get reset again on the next merge.

**Comment by sparr** [comment]
OK, that's all the skill and stat effects I could find. I need another set of eyes to go over the comments I've added and confirm that they appear to accurately describe what's going on in the code.

File: src/map.cpp [hunk 1]
```diff
@@ -5301,8 +5314,6 @@ void map::add_trap( const tripoint &p, const trap_id t)
 
 void map::disarm_trap( const tripoint &p )
 {
-    int skillLevel = g->u.skillLevel( skill_traps ); // TODO: same as below?
```
Comment (sparr) [inline src/map.cpp]: removing this redundant variable is out of scope for this PR, but also didn't seem to deserve its own PR.

**Comment by kevingranade** [comment]
Is this ready or stalled?  [CR] means "I don't know what to do and need feedback", not "please review for merge".  I'm not seeing anything objectionable in here...

**Comment by sparr** [comment]
> I need another set of eyes to go over the comments I've added and confirm that they appear to accurately describe what's going on in the code.

Someone, I thought it was you, on IRC said that this commenting needed to be looked over by a second set of eyes to make sure the comments actually match the code. I have tried a few times to convince someone on IRC to look over it. If you're happy with it, I can remove the [CR].

**Comment by sparr** [comment]
(and I'll do the merge conflict resolution if you give me a little warning when you want to merge it)

**Pull request merged**